### DISCUSSION TITLE
Hill Shading

### DISCUSSION THREAD

**Discussion by timokoesters** [organicmaps/organicmaps#576] [Ideas]
What do you think about adding a hill shading feature to make the shape of mountains more intuitive by showing their shadows?

Example from Locus: https://docs.locusmap.app/lib/exe/fetch.php?media=manual:user_guide:maps_settings:mapshade01.png

**Comment by biodranik** [comment]
It would be great, especially if someone from community help us to implement it :)

**Reply by reallew** [reply]
I do it. I volunteer. I always missed hill shading in maps.me.
I am a C++ (and Jav) dev with some experience in high performance graphics. And a map-loving traveler. And I have some free time for such a project. A perfect match, I would say.

**Reply by pastk** [reply]
That'll be great!!

**Comment by n-peugnet** [comment]
I wonder if it should be there by default or maybe be part of the "Terrain" layer ?

**Comment by not-available-username** [comment]
Is there any news regarding the development of this feature?

**Reply by biodranik** [reply]
Not yet. It will be implemented after refactoring the whole SRTM elevation data.

**Reply by skosch** [reply]
is it possible to help with this effort? Is there documentation on what needs to be done or how much progress has been made?

**Reply by biodranik** [reply]
Are you familiar with data compression algorithms and C++? Let's collaborate then )

**Reply by skosch** [reply]
I haven't touched either in about a decade, so I don't know if my contributions would be a net positive :stuck_out_tongue: but if it's something that can be done in chunks here and there, I may be able to give it a try (or at least help to document what needs to be done)
